src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -905,21 +905,21 @@ public TypeMirror getDeclaredType(Collection<TypeMirror> values,
      */
     public Set<TypeMirror> getAllInterfaces(TypeElement te) {
         Set<TypeMirror> results = new LinkedHashSet<>();
-        addSuperInterfaces(te.asType(), results);
+        addSuperInterfaces(te.asType(), results, new HashSet<>());
         return results;
     }
 
-    private void addSuperInterfaces(TypeMirror type, Set<TypeMirror> results) {
+    private void addSuperInterfaces(TypeMirror type, Set<TypeMirror> results, Set<Element> visited) {
         TypeMirror superType = null;
         for (TypeMirror t : typeUtils.directSupertypes(type)) {
             if (typeUtils.isSameType(t, getObjectType()))
                 continue;
             TypeElement e = asTypeElement(t);
             if (isInterface(e)) {
-                if (isPublic(e) || isLinkable(e)) {
+                if ((isPublic(e) || isLinkable(e)) && visited.add(typeUtils.asElement(t))) {
```

Comment:
Don't you always want to record you've visited the item, whether or not it is public or linkable?

Reply:
It would achieve the goal either way. However, your suggestion might look cleaner; I'll try it shortly.

Reply:
👍